dev-clean/src/main/scala/sai/lang/LLVM.scala

```diff
@@ -792,10 +792,9 @@ class MyVisitor extends LLVMParserBaseVisitor[LAST] {
     val floatStr = ctx.FLOAT_LIT.getText
     if (floatStr.contains('.')) FloatConst(floatStr.toFloat)
     else if (floatStr.startsWith("0x")) {
-      // Does not work, need to parse "0x3FF4CCCCC0000000" as 1.3
-      // val i = java.lang.Long.parseLong(floatStr.substring(2), 16)
-      // FloatConst(java.lang.Float.intBitsToFloat(i.intValue()))
-      ???
+      val hexString = floatStr.substring(2)
```

Comment:
It would be good to also add a test of this case into CI.

Reply:
Okay. I will look into that. Thanks for pointing it out!

Reply:
I think there is already a test at [here](https://github.com/Kraks/sai/blob/master/dev-clean/benchmarks/llvm/floatArith.c#L4). But many of those tests on llvm IR (including this one) don't really have asserts in them. We should probably do them later.

Reply:
sounds good to me.